src/ir/access_path_selectors.h

```diff
@@ -0,0 +1,91 @@
+#ifndef SRC_IR_ACCESS_PATH_SELECTORS_H_
+#define SRC_IR_ACCESS_PATH_SELECTORS_H_
+
+#include "src/ir/selector.h"
+
+#include <vector>
+
+#include "absl/hash/hash.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+namespace raksha::ir {
+
+// Represents an AccessPathSelectors descending through the members of types in a
+// Manifest. This is eventually used for printing edges to the resultant Datalog
+// file.
+//
+// We use this class instead of a bare vector for the following reasons:
+//
+// 1. Encapsulating the reversed vector prevents us from having to reason
+// about this all the time and potentially be confused by it.
+//
+// 2. Adding our own containing struct allows us to force copies to be
+// explicitly requested by making the copy constructor private, but
+// accessible with a public "copy" function. Because we are returning this
+// object owning variable-length memory in a bottom-up fashion, copies could
+// cause us to have quadratic performance.
+//
+// 3. Providing our own interface allows us to make this class immutable
+// except for move constructors. We define no "add", "insert", etc. methods;
+// "mutation" is done only by constructing a new object from a previous
+// object. That prevents many forms of state-manipulation bugs.
+//
+// 4. The name AccessPathSelectors is a bit more self-documenting.
+class AccessPathSelectors {
+ public:
+  // Create a leaf AccessPathSelectors from a single leaf selector.
+  explicit AccessPathSelectors(Selector leaf) {
+    reverse_selectors_.push_back(std::move(leaf));
```

Comment:
`emplace_back(...)` instead of `push_back(std::_move(...))`

Reply:
Discussed offline; `push_back` is better in most circumstances due to the possibility of C++ implicit constructor calls.